libraries/Adafruit_LittleFS/src/Adafruit_LittleFS_File.h

```diff
@@ -69,13 +75,21 @@ class File : public Stream
     uint32_t size (void);
 
     void close (void);
+
+    __attribute__((deprecated("Recommend use of isOpen() for clarity")))
     operator bool (void);
+
+    bool isOpen(void);
     char const* name (void);
 
     bool isDirectory (void);
     File openNextFile (uint8_t mode = FILE_O_READ);
     void rewindDirectory (void);
 
+    // Has to be public, in order to grant 'friend' permissions to Adafruit_LittleFS private member _mutex
```

Comment:
@henrygab as long as it works, maybe we should expose the LFS lock/unlock() file system to public :)

Reply:
Ok.

Because this PR fixes data corruption, I think it is important to accept even if you want to change some function names later.

Separately, if you want the design to better hide this, I would recommend following the PIMPL (Pointer-to-Implementation) solution.  This would allow 100% hiding of all the non-public APIs, and the interactions between the FS / File abstractions, without exposing any of this internal shared locking.  However, changing to PIMPL would be a bigger architectural change … my goal was to minimize changes while serializing access to LFS APIs.

Reply:
No, I don't need any fancy way to absolute hide API from user. I would simply add underscore to `_lock/_unlock` public API  with comment saying it is intended to be used internally  and expect user not to call it :D
https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/libraries/Bluefruit52Lib/src/bluefruit.h#L239